### PR TITLE
Pull my finger

### PR DESCRIPTION
**Pull request by gangster**

### CODE REVIEW

**Comment by zealoushacker** [comment]
But did you take a look at https://github.com/balanced/balanced-ruby/pull/6 and https://github.com/balanced/balanced-ruby/pull/7 before you had done some of the work herein?

File: lib/balanced.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
+$:.unshift File.join(File.dirname(__FILE__), "balanced", "resources")
```
Comment (ajsharp) [inline lib/balanced.rb:1]: Altering the load path should not be necessary here.

File: lib/balanced.rb [hunk 2]
```diff
@@ -17,34 +18,15 @@ module Balanced
...
-      self.client.post uri, data
-    end
-
-    def put uri, data = {}
-      self.client.put uri, data
-    end
-
-    def delete uri
-      self.client.delete uri
+      @client = Balanced::Client.new(api_key, @config.merge(options))
```
Comment (ajsharp) [inline lib/balanced.rb:25]: Object instances ftw.

File: lib/balanced/resources.rb [hunk 3]
```diff
@@ -1,419 +1,13 @@
...
+require "api_key"
+require "bank_account"
+require "card"
+require "credit"
+require "debit"
+require "hold"
+require "marketplace"
+require "merchant"
+require "refund"
+require "transaction"
```
Comment (ajsharp) [inline lib/balanced/resources.rb:13]: :sparkles:

File: lib/balanced/resources/account.rb [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        attributes['uri'] = Balanced::Marketplace.my_marketplace.send(self.class.collection_name + '_uri')
+      end
+      super attributes
+    end
+
+    # Returns a new Debit that represents a flow of money from this
+    # Account to your Marketplace.
+    #
+    # @return [Debit]
+    def debit (amount=nil,
```
Comment (ajsharp) [inline lib/balanced/resources/account.rb:22]: No space between method and and args list when using parens.

File: lib/balanced/resources/account.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      hold.save
+    end
+
+    # Returns a new Credit representing a transfer of funds from your
+    # Marketplace to this Account.
+    #
+    # @param [String] destination_uri A specific funding destination such as
+    #    a BankAccount already associated with this account.
+    # @return [Credit] A Credit representing the transfer of funds from
+    #    your Marketplace to this Account.
```
Comment (ajsharp) [inline lib/balanced/resources/account.rb:65]: Love the yardoc.

File: lib/balanced/resources/debit.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  # of capturing a Hold. If you create a Debit directly it will implicitly
+  # create the associated Hold if the funding source supports this.
+  #
+  # If no Hold is specified the Debit will by default be created using the
+  # most recently added funding source associated with the Account. You
+  # cannot change the funding source between creating a Hold and capturing
+  # it.
+  #
+  class Debit
+    include Balanced::Resource
```
Comment (ajsharp) [inline lib/balanced/resources/debit.rb:15]: I dig this approach. Nice.

File: lib/balanced/resources/resource.rb [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    end
+
+    # delegate the query to the pager module
+
+    def find uri
+      self.class.find uri
+    end
+
+    def save
+      uri = @attributes.delete('uri')
```
Comment (ajsharp) [inline lib/balanced/resources/resource.rb:16]: Why the delete here instead of just `@attributes['uri']`?

File: lib/balanced.rb (hunk 1 above)
Comment (gangster) [inline lib/balanced.rb:1]: Indeed.

File: lib/balanced/resources/resource.rb (hunk 7 above)
Comment (gangster) [inline lib/balanced/resources/resource.rb:16]: I was wondering the same thing.  Can probably be changed.  Worth investigating...

**Comment by mahmoudimus** [comment]
Thanks!!

**Pull request closed without merging**